src/librbd/api/Mirror.cc

```diff
@@ -2170,6 +2490,1453 @@ void Mirror<I>::image_snapshot_create(I *ictx, uint32_t flags,
+int are_images_mirror_disabled(std::vector<I *> *image_ctxs) {
+  std::vector<mirror_image_info_t> mirror_images(image_ctxs->size());
+  std::vector<C_SaferCond> on_finishes(image_ctxs->size());
+  int ret_code = 1;
+  ImageCtx *ictx;
+  for (size_t i = 0; i < image_ctxs->size(); i++) {
+    ictx = (*image_ctxs)[i];
+    librbd::api::Mirror<>::image_get_info(ictx,
+                                          &mirror_images[i], &on_finishes[i]);
+  }
+
+  for (size_t i = 0; i < image_ctxs->size(); i++) {
+    int r = on_finishes[i].wait();
+    if (r < 0) {
+      if (ret_code >= 0) {
+        ret_code = r;
+      }
+    } else if (mirror_images[i].state != RBD_MIRROR_IMAGE_DISABLED) {
+      ictx = (*image_ctxs)[i];
+      lderr(ictx->cct) << "image " << ictx->name
+                       << " is not disabled for mirroring" << dendl;
+      if (ret_code == 1) {
+        ret_code = 0;
+      }
+    }
+  }
+
+  return ret_code;
+}
+
+template <typename I>
+int prepare_group_images(IoCtx& group_ioctx,
+                         std::string group_id,
+                         cls::rbd::MirrorGroupState group_state,
+                         std::vector<I *> *image_ctxs,
+                         cls::rbd::GroupSnapshot *group_snap,
+                         std::vector<uint64_t> &quiesce_requests,
+                         cls::rbd::MirrorSnapshotState snap_state,
+                         std::set<std::string> *mirror_peer_uuids,
+                         uint64_t internal_flags) {
+  CephContext *cct = (CephContext *)group_ioctx.cct();
+  ldout(cct, 20) << dendl;
+
+  librados::IoCtx default_ns_ioctx;
+  default_ns_ioctx.dup(group_ioctx);
+  default_ns_ioctx.set_namespace("");
+  std::vector<cls::rbd::MirrorPeer> peers;
+  int r = cls_client::mirror_peer_list(&default_ns_ioctx, &peers);
+  if (r < 0) {
+    lderr(cct) << "error reading mirror peers: " << cpp_strerror(r) << dendl;
+    return r;
+  }
+
+  for (auto &peer : peers) {
+    if (peer.mirror_peer_direction == cls::rbd::MIRROR_PEER_DIRECTION_RX) {
+      continue;
+    }
+    mirror_peer_uuids->insert(peer.uuid);
+  }
+
+  if (mirror_peer_uuids->empty()) {
+    lderr(cct) << "no mirror tx peers configured for the pool" << dendl;
+    return -EINVAL;
+  }
+  r = open_group_images(group_ioctx, group_id, image_ctxs);
+  if (r < 0) {
+    return r;
+  }
+
+  if (group_state != cls::rbd::MIRROR_GROUP_STATE_ENABLED) {
+    auto group_pool_id = group_ioctx.get_id();
+    for (auto image_ctx: *image_ctxs) {
+      if (group_pool_id != image_ctx->md_ctx.get_id()) {
+        lderr(cct) << "cannot enable mirroring: image is in a different pool"
+                   << dendl;
+        return -ENOTSUP;
+      }
+    }
+
+    r = are_images_mirror_disabled(image_ctxs);
+    if (r == 0) { // false
+      lderr(cct) << "image(s) in group is mirror enabled" << dendl;
+      return -EINVAL;
+    } else if (r < 0) {
+      lderr(cct) << "failed to validate image(s) in group is mirror enabled: "
```

Comment:
As already suggested here,
https://github.com/ceph/ceph/pull/62945#discussion_r2159180778

"failed to validate image(s) disabled for mirroring" ?